### PR TITLE
Add it.mijorus.smile

### PR DESCRIPTION
**Pull request by mijorus**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:** https://github.com/mijorus/smile
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: it.mijorus.smile.json [hunk 1]
```diff
@@ -0,0 +1,42 @@
+{
+    "app-id" : "it.mijorus.smile",
+    "runtime" : "org.gnome.Platform",
+    "runtime-version" : "41",
+    "sdk" : "org.gnome.Sdk",
+    "command" : "smile",
+    "finish-args" : [
+        "--share=network",
+        "--share=ipc",
+        "--socket=x11",
```
Comment (hfiguiere) [inline it.mijorus.smile.json:10]:
```suggestion
        "--socket=fallback-x11",
```

File: it.mijorus.smile.json [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    "modules" : [
+        "python3-requirements.json",
+        {
+            "name" : "smile",
+            "builddir" : true,
+            "buildsystem" : "meson",
+            "sources" : [
+                {
+                    "type" : "dir",
+                    "path" : "./"
```
Comment (hfiguiere) [inline it.mijorus.smile.json:34]: you need to get the source here from upstream.

File: it.mijorus.smile.json [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+                }
+            ]
+        }
+    ],
+    "build-options" : {
+        "env" : {        }
+    }
```
Comment (hfiguiere) [inline it.mijorus.smile.json:41]:
```suggestion
    ]
```

File: it.mijorus.smile.json [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+        "/man",
+        "/share/doc",
+        "/share/gtk-doc",
+        "/share/man",
+        "/share/pkgconfig",
+        "*.la",
+        "*.a"
+    ],
+    "modules" : [
+        "python3-requirements.json",
```
Comment (hfiguiere) [inline it.mijorus.smile.json:26]: this is missing

**Review by hfiguiere — CHANGES_REQUESTED** [review]

**Comment by mijorus** [comment]
Hi, sorry this is my first time uploading here. I have fixed the reported errros.

**Comment by barthalion** [comment]
bot, build it.mijorus.smile

**Comment by hfiguiere** [comment]
AppData fail to validate

```
builddir/files/share/appdata/it.mijorus.smile.appdata.xml: FAILED:
• tag-missing           : <content_rating> required [use https://odrs.gnome.org/oars]
• style-invalid         : Not enough <screenshot> tags, minimum is 1
• tag-missing           : <release> required
• tag-missing           : <name> is not present
• tag-missing           : <summary> is not present
• tag-missing           : <url> is not present
Validation of files failed
```

**Comment by mijorus** [comment]
bot, build it.mijorus.smile

**Comment by mijorus** [comment]
> Started [test build 77313](https://flathub.org/builds/#/builders/32/builds/77313)

Did it crash?

**Comment by mijorus** [comment]
bot, build it.mijorus.smile

**Comment by mijorus** [comment]
bot, build it.mijorus.smile

**Comment by mijorus** [comment]
I tested it in various VMs, everything seems to work fine

**Comment by barthalion** [comment]
It doesn't seem to copy emojis for me on Fedora 35 Silverblue.

**Comment by mijorus** [comment]
> It doesn't seem to copy emojis for me on Fedora 35 Silverblue.

It had more to do with my code than with Fedora apparently 🤣😂😅

**Comment by mijorus** [comment]
bot, build it.mijorus.smile

**Comment by mijorus** [comment]
> Build [77519 successful](https://flathub.org/builds/#/builders/32/builds/77519) To test this build, install it from the testing repository:
> 
> ```
> flatpak install --user https://dl.flathub.org/build-repo/75387/it.mijorus.smile.flatpakref
> ```

This new build seems to work just fine in my Fedora 35 VM

**Pull request closed without merging**